### PR TITLE
[Java] Add killActor API in Java

### PR DESCRIPTION
**Pull request by kfstorm**
## Why are these changes needed?

Support `killActor` in Java worker based on https://github.com/ray-project/ray/pull/6523.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://ray.readthedocs.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.

### CODE REVIEW

**Review by edoakes — APPROVED** [review]
Awesome! :)

File: src/ray/core_worker/lib/java/org_ray_runtime_RayNativeRuntime.cc [hunk 1]
```diff
@@ -133,6 +133,13 @@ JNIEXPORT void JNICALL Java_org_ray_runtime_RayNativeRuntime_nativeSetResource(
   THROW_EXCEPTION_AND_RETURN_IF_NOT_OK(env, status, (void)0);
 }
 
+JNIEXPORT void JNICALL Java_org_ray_runtime_RayNativeRuntime_nativeKillActor(
+    JNIEnv *env, jclass, jlong nativeCoreWorkerPointer, jbyteArray actorId) {
+  auto core_worker = reinterpret_cast<ray::CoreWorker *>(nativeCoreWorkerPointer);
+  auto status = core_worker->KillActor(JavaByteArrayToId<ActorID>(env, actorId));
+  THROW_EXCEPTION_AND_RETURN_IF_NOT_OK(env, status, (void)0);
```
Comment (jovany-wang) [inline src/ray/core_worker/lib/java/org_ray_runtime_RayNativeRuntime.cc]:
```suggestion
  THROW_EXCEPTION_AND_RETURN_IF_NOT_OK(env, status, (void) 0);
```

**Review by jovany-wang — APPROVED** [review]
LGTM. (ps: I left 2 minor comments about code style.)

File: java/runtime/src/main/java/org/ray/runtime/RayNativeRuntime.java [hunk 2]
```diff
@@ -127,6 +129,14 @@ public void setResource(String resourceName, double capacity, UniqueId nodeId) {
     nativeSetResource(nativeCoreWorkerPointer, resourceName, capacity, nodeId.getBytes());
   }
 
+  @Override
+  public void killActor(RayActor<?> actor) {
+    if (!((NativeRayActor)actor).isDirectCallActor()) {
+      throw new UnsupportedOperationException("Only direct call actors can be killed.");
```
Comment (raulchen) [inline java/runtime/src/main/java/org/ray/runtime/RayNativeRuntime.java]: Why only direct call actor?

**Review by raulchen — APPROVED** [review]

**Comment by raulchen** [comment]
CI failure is related

File: java/test/src/main/java/org/ray/api/TestUtils.java [hunk 3]
```diff
@@ -28,8 +28,13 @@ public static void skipTestUnderSingleProcess() {
   }
 
   public static void skipTestIfDirectActorCallEnabled() {
-    if (ActorCreationOptions.DEFAULT_USE_DIRECT_CALL) {
-      throw new SkipException("This test doesn't work when direct actor call is enabled.");
+    skipTestIfDirectActorCallEnabled(true);
+  }
+
+  public static void skipTestIfDirectActorCallEnabled(boolean enabled) {
```
Comment (zhijunfu) [inline java/test/src/main/java/org/ray/api/TestUtils.java]: Maybe this is more clear?

```suggestion
  public static void skipTestIfDirectActorCallDisabled() {
```

**Review by zhijunfu — APPROVED** [review]

File: src/ray/core_worker/lib/java/org_ray_runtime_RayNativeRuntime.cc (hunk 1 above)
Comment (kfstorm) [reply src/ray/core_worker/lib/java/org_ray_runtime_RayNativeRuntime.cc]: clang-format suggest the the whitespace should be removed.

**Pull request merged**